### PR TITLE
ENG-3768: Add access package review tab to admin UI

### PR DESCRIPTION
**Pull request by JadeCara**
Ticket [ENG-3768]

### Description Of Changes

Port of Jason's access package review tab from `demo/april-access-packages` into a flag-gated implementation. The tab allows admins to review access request data organized by data use, apply per-field redactions, preview/download the package as a ZIP, and approve to resume the pipeline.

**Gated behind:** `flags.accessPackages` (frontend flag, `test: false`) AND `useFeatures().plus` (fidesplus required). Nothing renders unless both are true.

> **Dependency:** Requires the fidesplus routes PR on `ENG-3768/access-package-routes` to be merged for the API endpoints to exist.

### Code Changes

**New files:**
* `access-package/PrivacyRequestDetailsAccessPackageTab.tsx` — full review tab with data-use collapse panels, per-field redaction checkboxes, approve button, ZIP download
* `access-package/access-package.slice.ts` — RTK Query endpoints for all 4 access package API routes (get, redact, approve, download)
* `access-package/types.ts` — local types mirroring fidesplus Pydantic schemas (will be replaced with generated types once fidesplus routes are merged)
* `access-package/utils.ts` — redaction key mapping and value rendering helpers
* `access-package/utils.test.ts` — 14 unit tests for utils
* `cypress/fixtures/privacy-requests/access-package.json` — Cypress test fixture

**Modified files:**
* `api.slice.ts` — register `"Access Package"` RTK Query tag type
* `PrivacyRequest.tsx` — inject access package tab, gated by `flags.accessPackages && hasPlus`, auto-selects when status is `awaiting_access_review`
* `helpers.ts` — add `BulkActionType.APPROVE` for `awaiting_access_review` status
* `types.ts` — add `ACCESS_PACKAGE_APPROVED` execution log status + `access_review_approved_at/by` fields on `PrivacyRequestEntity`
* `EventLog.tsx` — add `ACCESS_PACKAGE_APPROVED` to audit statuses with details
* `privacy-requests.cy.ts` — Cypress tests for the access package tab

### Steps to Confirm

1. Enable the `accessPackages` flag in `flags.json` (`development: true` is already set)
2. With fidesplus running, navigate to a privacy request in `awaiting_access_review` status
3. Verify the "Access package" tab auto-activates and shows data organized by data use
4. Verify stats card shows correct field/system/redacted counts
5. Uncheck a field row → verify PUT to `/redactions` fires with correct payload
6. Click "Approve & continue" → verify POST to `/approve` fires and success message shows
7. Click "Preview" → verify GET to `/download` fires and ZIP downloads
8. With flag disabled, verify the tab does not appear
9. On a request NOT in `awaiting_access_review`, verify the tab is disabled

### Pre-Merge Checklist

* [x] Issue requirements met
* [ ] All CI pipelines succeeded
* [x] `CHANGELOG.md` updated
  * [ ] Add a https://github.com/ethyca/fides/labels/db-migration label to the entry if your change includes a DB migration
  * [ ] Add a https://github.com/ethyca/fides/labels/high-risk label to the entry if your change includes a high-risk change (i.e. potential for performance impact or unexpected regression) that should be flagged
  * [ ] Updates unreleased work already in Changelog, no new entry necessary
* UX feedback:
  * [ ] All UX related changes have been reviewed by a designer
  * [x] No UX review needed
* Followup issues:
  * [ ] Followup issues created
  * [x] No followup issues
* Database migrations:
  * [ ] Ensure that your downrev is up to date with the latest revision on `main`
  * [ ] Ensure that your `downgrade()` migration is correct and works
  * [x] No migrations
* Documentation:
  * [ ] Documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] Documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
  * [x] No documentation updates required

[ENG-3768]: https://ethyca.atlassian.net/browse/ENG-3768?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by rayharnett** [comment]
# Code Review: PR #8231 — ENG-3768: Add access package review tab to admin UI

**Base branch:** `ENG-3768/access-package-hooks`
**Head branch:** `ENG-3768/access-package-frontend`
**Changes:** +904 / −3 lines, 13 files

---

## 🚨 Critical Issues (Must Fix)

### 1. Race condition: stale `data` used during `updateRedactions` loading state

**Files:** `PrivacyRequestDetailsAccessPackageTab.tsx`, lines 123–145

The `handleCategorySelectionChange` callback captures `data` from the closure at render time. When `updateRedactions` is running (`isSaving === true`), the RTK Query cache may have been invalidated (by `approveAccessPackage` or a refetch), but the callback still references the **stale** `data` object from when it was created — not the current cache. This means:

- If a user quickly unchecks multiple categories, each callback closes over the same stale `data`.
- The preserved redactions computed from stale `data` may drop a previously saved redaction.

**Fix:** Use the RTK Query cache selector (`useSelector` / `api.useQueryState`) inside the handler to read the current cached response, or disable the table rows while `isSaving` (which is already done via `getCheckboxProps: ({ disabled })` in `CategoryTable`). The simplest fix is to add a guard at the top of `handleCategorySelectionChange`:

```ts
if (isSaving) {
  return; // Avoid stale-closure writes during in-flight mutations
}
```

### 2. Blob download silently fails on non-Blob responses from the backend

**File:** `PrivacyRequestDetailsAccessPackageTab.tsx`, lines 152–167

The `handleDownload` handler checks `blob.size === 0`, but if the backend returns an error JSON (e.g., `{ "detail": "not ready" }`) with `content-type: application/json`, calling `.blob()` on it returns a valid Blob containing the JSON string. `blob.size` will be non-zero (the length of the error JSON), so the check passes and a `.zip` file containing a JSON string gets downloaded.

**Fix:** Check the response `content-type` header before calling `.blob()`, or inspect the blob content:

```ts
const contentType = result.response?.headers?.get("content-type");
if (!contentType?.includes("application/zip")) {
  message.error("The server did not return a ZIP file.");
  return;
}
```

This requires the RTK Query endpoint to expose the raw response headers. One approach: use a custom `transformResponse` that returns `{ data: Blob, contentType: string }`.

### 3. Missing error handling for `approve` mutation — no retry or user feedback on partial failure

**File:** `PrivacyRequestDetailsAccessPackageTab.tsx`, lines 169–175

`handleApprove` calls `approve(privacyRequestId)` and checks `isErrorResult`. However, if the backend returns a 202 (partial success from the migration endpoint behavior we documented in memory), this code would incorrectly treat it as an error. More broadly, there's no loading shadow or double-click protection — a user can click "Approve" repeatedly before the response arrives, potentially creating duplicate approval requests.

**Fix:** Add double-click protection via a `hasBeenApproved` ref or by disabling the button during `isApproving` (already done) AND showing a confirmation toast that doesn't re-close until the response arrives.

---

## ⚠️ Warnings (Should Fix)

### 4. `redactionKey` collision: empty string and null field_path produce identical keys

**File:** `clients/admin-ui/src/features/privacy-requests/access-package/utils.ts`, line 10

The test at `utils.test.ts` line 35–38 documents this: `redactionKey("s", 1, "")` and `redactionKey("s", 1, null)` produce `"s::1::"`. The comment says "intentional" because "the API never produces an empty-string field_path for REDACT redactions, only null for REMOVE_RECORD."

This is a fragile invariant. If the API ever changes or if an entry has `field_path: ""`, the collision will cause incorrect redaction matching in `handleCategorySelectionChange` (line 127: `categoryKeys.has(k)`).

**Suggestion:** Use a separator that cannot appear in the source string, e.g., `|`, or use a structured key like `{source, recordIndex, fieldPath}` as the Table `rowKey` instead of a string join.

### 5. `useLazyDownloadAccessPackageQuery` — no cache invalidation on success or error

**File:** `clients/admin-ui/src/features/privacy-requests/access-package/access-package.slice.ts`, lines 54–58

The `downloadAccessPackage` query uses `cache: "no-cache"` (correct), but it does not `invalidatesTags` on either success or error. This means:

- If the download succeeds but the backend has a long-running ZIP build, subsequent downloads will not re-fetch.
- The query result is never removed from Redux state after the download completes, causing memory leak for users who navigate between many privacy requests.

**Fix:** Add `transformResponse` that resolves the query, and dispatch a manual cache eviction or add an `invalidatesTags` via a mutation wrapper instead of a query.

### 6. No optimistic update on redaction toggle — UI lags behind server

**File:** `PrivacyRequestDetailsAccessPackageTab.tsx`, lines 123–145

When a user unchecks a row, the UI only updates after the server responds. For large packages (hundreds of fields), this can feel sluggish. If the server fails, the UI reverts — which is correct — but there's no loading indicator on individual rows.

**Suggestion:** Add an optimistic update that immediately marks the entry as `redacted: true` in a local state slice, then reverts on error. This is a nice-to-have UX improvement.

### 7. `PrivacyRequest.tsx` — tab key ordering depends on array position, not semantic priority

**File:** `clients/admin-ui/src/features/privacy-requests/PrivacyRequest.tsx`, lines 75–81

The access package tab is appended to the end of the `items` array. If future tabs are added, an admin might not discover it without knowing to look at the end. Consider placing it before "Activity" since `awaiting_access_review` is an in-progress state that requires action.

---

## 💡 Suggestions (Improvements)

### 8. Type safety: local types will diverge from fidesplus schema

**File:** `clients/admin-ui/src/features/privacy-requests/access-package/types.ts`

The comment says "These mirror the fidesplus Pydantic schemas" and "can be replaced with generated imports." Consider generating these types now via an OpenAPI spec from fidesplus (even if merged to a feature branch) rather than maintaining them manually. Manual mirrors are the #1 source of runtime errors in this codebase (see memory about the bloomreach migration script).

### 9. Cypress fixture hardcodes data that should be parameterized

**File:** `clients/admin-ui/cypress/fixtures/privacy-requests/access-package.json`

The fixture uses specific values (`"Jane Smith"`, `"jane@example.com"`) that are not referenced by the tests. The tests check for "Fields: 3", "Systems: 2" counts but not specific values. Consider making the fixture data-driven via a factory function so tests can assert on specific field paths without baking values into fixtures.

### 10. Missing `testId` on the main content container

The root `<Flex>` has `data-testid="access-package-tab"` but the individual stat cards and section headers do not. The Cypress test at line 104–113 uses `.ant-statistic` and `.ant-statistic-content-value` selectors which are fragile — they rely on fidesui's internal CSS class naming. Consider adding `data-testid` attributes to the stat values for more resilient tests.

### 11. `CHANGELOG.md` entry is minimal but acceptable

The changelog fragment (`changelog/8231-access-package-review-tab.yaml`) is a single-line description. Consider adding labels `[feature, frontend]` for discoverability in changelog generation.

---

## ✅ Good Practices

1. **Feature flag gating is thorough** — `PrivacyRequest.tsx` correctly gates the tab behind both `flags.accessPackages` AND `hasPlus`, and handles all three states: flag off, plus unavailable, status not matching.

2. **RTK Query integration is clean** — The `access-package.slice.ts` properly defines tag-based invalidation (`"Access Package"` + `"Request"`) for cache coherence. The `responseHandler` for the download endpoint correctly handles blob streaming.

3. **Unit test coverage on pure functions is excellent** — `utils.test.ts` (14 tests) thoroughly covers `redactionKey`, `rowKeyFor`, `entryToRedaction`, and `renderValue` including edge cases (circular refs, whitespace preservation, null/undefined handling).

4. **Cypress E2E tests cover all user flows** — The fixture data is well-structured, and the tests cover: flag disabled (tab hidden), wrong status (tab disabled), correct status (tab active), redaction toggle, approve flow, and download flow.

5. **Error handling is consistent** — All three async operations (get, update, approve, download) use `isErrorResult` pattern from the existing codebase for uniform error display.

6. **The `PrivacyRequestDetailsAccessPackageTab` component is well-structured** — It separates pure rendering (`CategoryTable`, `renderSectionBody/Label`) from async logic hooks, making it testable and readable.

7. **Appropriate skeleton/empty/error states** — The component handles `isLoading`, `error`, and empty data gracefully with `<Skeleton>`, `<Alert type="error">`, and `<Empty>` components.

8. **Safari-compatible blob download** — The `setTimeout(() => window.URL.revokeObjectURL(url), 0)` pattern correctly handles Safari's asynchronous download behavior.

File: clients/admin-ui/src/features/privacy-requests/helpers.ts [hunk 1]
```diff
@@ -53,7 +53,10 @@ const AVAILABLE_ACTIONS_BY_STATUS: Record<
     BulkActionType.DENY,
     BulkActionType.DELETE,
   ],
-  [PrivacyRequestStatus.AWAITING_ACCESS_REVIEW]: [BulkActionType.DELETE],
+  [PrivacyRequestStatus.AWAITING_ACCESS_REVIEW]: [
+    BulkActionType.APPROVE,
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-requests/helpers.ts]: Good catch. Removed `BulkActionType.APPROVE` from `AWAITING_ACCESS_REVIEW`. A bulk approve would bypass the per-request redaction review, which defeats the purpose of this tab. Approval has to go through the access-package tab one request at a time.

File: clients/admin-ui/src/features/privacy-requests/helpers.ts [hunk 2]
```diff
@@ -53,7 +53,10 @@ const AVAILABLE_ACTIONS_BY_STATUS: Record<
     BulkActionType.DENY,
     BulkActionType.DELETE,
   ],
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-requests/helpers.ts:55]: Confirmed. Fixed by removing `BulkActionType.APPROVE` from `AWAITING_ACCESS_REVIEW` in `helpers.ts`. See thread on line 57.

File: clients/admin-ui/src/features/privacy-requests/access-package/utils.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * Stable key for a redactable target. Used as the Table rowKey for
+ * AccessPackageEntry rows and to compare against existing RedactionEntry
+ * items (whose field_path is null for REMOVE_RECORD-type redactions).
+ */
+export const redactionKey = (
+  source: string,
+  recordIndex: number,
+  fieldPath: string | null | undefined,
+): string => `${source}::${recordIndex}::${fieldPath ?? ""}`;
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-requests/access-package/utils.ts]: Switched `redactionKey` to `JSON.stringify([source, recordIndex, fieldPath ?? null])`. That disambiguates null vs. undefined vs. empty-string field_path and removes the `::` separator collision risk for sources that contain `::`. Updated the tests in `utils.test.ts` to assert the new encoding and to verify both collision cases.

File: clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+    );
+    data.other?.categories.forEach((cat) => entries.push(...cat.entries));
+    return entries;
+  }, [data]);
+
+  const totalFields = allEntries.length;
+  const redactedCount = allEntries.filter((e) => e.redacted).length;
+  const systemCount = useMemo(
+    () => new Set(allEntries.map((e) => e.system || e.source)).size,
+    [allEntries],
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx:145]: Fixed. `systemCount` now uses the same `system_name || system || source` resolution as the column renderer.

File: clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+    }
+    const url = window.URL.createObjectURL(blob);
+    const a = document.createElement("a");
+    a.href = url;
+    a.download = `access-package-${privacyRequestId}.zip`;
+    a.click();
+    a.remove();
+    // Defer revoke so Safari has time to start the download.
+    setTimeout(() => window.URL.revokeObjectURL(url), 0);
+  }, [downloadPackage, privacyRequestId, message]);
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx:208]: Bumped to 100ms for Safari.

File: clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+          result.error,
+          "A problem occurred while downloading the access package.",
+        ),
+      );
+      return;
+    }
+    const blob = result.data;
+    if (!(blob instanceof Blob) || blob.size === 0) {
+      message.error(
+        "The server did not return a ZIP file. The access package may not be ready yet.",
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx:194]: Bumped to 100ms. See comment on line 203.

File: clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx [hunk 7]
```diff
@@ -0,0 +1,375 @@
...
+            onSelectionChange={handleCategorySelectionChange}
+            disabled={!isAwaitingReview}
+          />
+        </div>
+      ))}
+    </Flex>
+  );
+
+  const renderSectionLabel = (
+    name: string,
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx]: Wrapped `renderSectionBody` and `renderSectionLabel` in `useCallback` (lifted above the early returns to satisfy rules-of-hooks). The `sections` array is still rebuilt per render since it depends on `data`, but the inner callbacks are now stable.

File: clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx [hunk 8]
```diff
@@ -0,0 +1,375 @@
...
+    categories: AccessPackageCategory[],
+  ) => (
+    <Space>
+      <Typography.Text strong>{name}</Typography.Text>
+      <Typography.Text type="secondary">
+        ({categories.reduce((n, c) => n + c.entries.length, 0)} fields)
+      </Typography.Text>
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx]: Fixed. Now renders `({count} {count === 1 ? "field" : "fields"})`. Updated the Cypress assertion at line 800 to match.

File: clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx [hunk 9]
```diff
@@ -0,0 +1,375 @@
...
+          <Typography.Title level={3} className="pb-2">
+            {cat.name}
+          </Typography.Title>
+          <CategoryTable
+            category={cat}
+            onSelectionChange={handleCategorySelectionChange}
+            disabled={!isAwaitingReview}
+          />
+        </div>
+      ))}
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx]: Fixed. See the thread on line 269.

File: clients/admin-ui/cypress/e2e/privacy-requests.cy.ts [hunk 10]
```diff
@@ -719,4 +719,137 @@ describe("Privacy Requests", () => {
...
+            .should("contain.text", "2");
+          cy.contains(".ant-statistic", "Redacted")
+            .find(".ant-statistic-content-value")
+            .should("contain.text", "1");
+        });
+
+        cy.contains("Essential Service Operations").should("be.visible");
+        cy.contains("Other Data").should("be.visible");
+        cy.contains("(2 fields)").should("be.visible");
+        cy.contains("(1 fields)").should("be.visible");
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/cypress/e2e/privacy-requests.cy.ts]: Updated the assertion to `(1 field)` to match the fixed pluralization.

File: clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx [hunk 11]
```diff
@@ -0,0 +1,375 @@
...
+
+  return (
+    <Flex vertical gap="large" data-testid="access-package-tab">
+      {!isAwaitingReview && (
+        <Alert
+          type="info"
+          showIcon
+          title="This request is not currently awaiting access review."
+          description="Redactions and approval are only available while the request is in awaiting_access_review."
+        />
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx:316]: Reworded to "Redactions and approval are only available while the request is awaiting access review."

File: clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx [hunk 12]
```diff
@@ -0,0 +1,375 @@
...
+
+      {isEmpty && <Empty description="Access package is empty" />}
+
+      {sections.length > 0 && (
+        <Collapse defaultActiveKey={allSectionKeys} items={sections} />
+      )}
+
+      {attachments.length > 0 && (
+        <div>
+          <Typography.Title level={5}>Attachments</Typography.Title>
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx:366]: Swapped the wrapping `<div>` for `<Flex vertical gap="small">`.

File: clients/admin-ui/src/features/privacy-requests/access-package/access-package.slice.ts [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+      ],
+    }),
+
+    downloadAccessPackage: build.query<Blob, string>({
+      query: (privacyRequestId) => ({
+        url: `plus/privacy-request/${privacyRequestId}/access-package/download`,
+        responseHandler: (response: Response) => response.blob(),
+        cache: "no-cache",
+      }),
+    }),
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-requests/access-package/access-package.slice.ts:52]: Converted `downloadAccessPackage` from `build.query` to `build.mutation`. Mutations aren't subscribed to a cache key, so they execute every time the trigger fires. Updated the consumer to use `useDownloadAccessPackageMutation`.

File: clients/admin-ui/src/features/privacy-requests/access-package/access-package.slice.ts [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+        { type: "Access Package", id: privacyRequestId },
+        "Request",
+      ],
+    }),
+
+    downloadAccessPackage: build.query<Blob, string>({
+      query: (privacyRequestId) => ({
+        url: `plus/privacy-request/${privacyRequestId}/access-package/download`,
+        responseHandler: (response: Response) => response.blob(),
+        cache: "no-cache",
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-requests/access-package/access-package.slice.ts:50]: Switched to a mutation. See thread on line 54.

File: clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx [hunk 15]
```diff
@@ -0,0 +1,375 @@
...
+        description={getErrorMessage(error)}
+      />
+    );
+  }
+
+  if (!data) {
+    return <Empty description="No access package available" />;
+  }
+
+  const { attachments } = data;
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-requests/access-package/PrivacyRequestDetailsAccessPackageTab.tsx:279]: Tightened `isEmpty` to also treat `data.other` with zero categories as empty.

File: clients/admin-ui/src/features/privacy-requests/types.ts [hunk 16]
```diff
@@ -133,6 +136,8 @@ export interface PrivacyRequestEntity {
   source?: string;
   location?: string;
   property_id?: string | null;
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-requests/types.ts:138]: Removed `access_review_approved_at` and `access_review_approved_by` from `PrivacyRequestEntity`. They aren't referenced anywhere yet; they can come back when a consumer needs them.

File: clients/admin-ui/src/features/privacy-requests/access-package/types.ts [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+  REDACT = "redact",
+  REMOVE_FIELD = "remove_field",
+  REMOVE_RECORD = "remove_record",
+}
+
+export interface RedactionEntry {
+  source: string;
+  record_index: number;
+  field_path: string | null;
+  type: RedactionType;
```
[the remark this answers is not in the document]
Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-requests/access-package/types.ts]: Added a doc comment on `RedactionEntry.field_path` explaining that null is only produced for `REMOVE_RECORD` entries (which this UI doesn't emit, but must round-trip).

**Review by gilluminate — APPROVED** [review]
Ship it! 🚀
